helix-term/src/application.rs

```diff
@@ -87,7 +88,7 @@ impl Application {
 
         let mut editor = Editor::new(size, theme_loader.clone(), syn_loader.clone());
 
-        let mut editor_view = Box::new(ui::EditorView::new(config.keymaps.clone()));
+        let mut editor_view = Box::new(ui::EditorView::new(std::mem::take(&mut config.keys)));
```

Comment:
this is equivalent to `config.keys.take()`.

Reply:
I don't think that works but let me try tonight.

Reply:
https://doc.rust-lang.org/src/core/option.rs.html#965

Reply:
But it's not an `Option` here.

Reply:
Ah I see, you're swapping it for the default value.

Reply:
```
error[E0599]: the method `take` exists for struct `Keymaps`, but its trait bounds were not satisfied
   --> helix-term/src/application.rs:91:72
    |
91  |         let mut editor_view = Box::new(ui::EditorView::new(config.keys.take()));
    |                                                                        ^^^^ method cannot be called on `Keymaps` due to unsatisfied trait bounds
    | 
   ::: helix-term/src/keymap.rs:141:1
    |
141 | pub struct Keymaps(pub HashMap<Mode, HashMap<KeyEvent, Command>>);
    | ------------------------------------------------------------------
    | |
    | method `take` not found for this
    | doesn't satisfy `Keymaps: Iterator`
```